### PR TITLE
Fix metric round at epoch start

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed MetricRoundAtEpochStart reset issue after bootstrap
- proper fix for app status handler
- refactoring on the metrics area, added unit tests, removed redundant mocks

Testing scenario: start an #allIn testnet, in epoch 0, before upgrade, take one node, stop it, start it and connect to the termui interface. You should observe bad termui epoch estimation like the following image:
temp1
After the epoch changes, the termui will show the correct info in that area.
After the upgrade to this branch, repeat the process: take one node, stop it, start it and connect to the termui interface. You should observe the correct termui epoch estimation like this:
temp2

### CODE REVIEW

File: node/metrics/metrics.go [hunk 1]
```diff
@@ -21,6 +24,104 @@ type StatusHandlersUtils interface {
...
+	appStatusHandler.SetStringValue(common.MetricP2PUnknownPeers, initString)
+
+	appStatusHandler.SetStringValue(common.MetricInflation, initZeroString)
+	appStatusHandler.SetStringValue(common.MetricDevRewardsInEpoch, initZeroString)
+	appStatusHandler.SetStringValue(common.MetricTotalFees, initZeroString)
+
+	return nil
+}
+
+// InitConfigMetrics will init the configs from epoch config
```
Comment (bogdan-rosianu) [inline node/metrics/metrics.go:78]: will init the enable epochs configuration metrics from epoch config

File: process/sync/loadPersistentMetrics.go [hunk 2]
```diff
@@ -90,7 +90,6 @@ func prepareMetricMaps(metricsMap map[string]interface{}) (map[string]uint64, ma
 	uint64Map[common.MetricCountLeader] = persister.GetUint64(metricsMap[common.MetricCountLeader])
 	uint64Map[common.MetricNumProcessedTxs] = persister.GetUint64(metricsMap[common.MetricNumProcessedTxs])
 	uint64Map[common.MetricNumShardHeadersProcessed] = persister.GetUint64(metricsMap[common.MetricNumShardHeadersProcessed])
-	uint64Map[common.MetricRoundAtEpochStart] = persister.GetUint64(metricsMap[common.MetricRoundAtEpochStart])
 	uint64Map[common.MetricNonceAtEpochStart] = persister.GetUint64(metricsMap[common.MetricNonceAtEpochStart])
```
Comment (bogdan-rosianu) [inline process/sync/loadPersistentMetrics.go:93]: maybe treat `MetricNonceAtEpochStart` the same way as the one for round?

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: common/constants.go [hunk 3]
```diff
@@ -427,8 +427,47 @@ const (
 	//MetricDelegationSmartContractEnableEpoch represents the epoch when delegation smart contract is enabled epoch should not be 0
 	MetricDelegationSmartContractEnableEpoch = "erd_delegation_smart_contract_enable_epoch"
 
+	// MetricCorrectLastUnjailedEnableEpoch represents the epoch when the correction applied on the last unjailed node has been applied
+	MetricCorrectLastUnjailedEnableEpoch = "erd_correct_last_unjailed_enable_epoch"
+
+	// MetricBalanceWaitingListsEnableEpoch represents the epoch when the balance waiting lists on shards has been applied
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply common/constants.go:433]: fixed

File: process/sync/loadPersistentMetrics.go (hunk 2 above)
Comment (iulianpascalau) [reply process/sync/loadPersistentMetrics.go:93]: done

File: node/metrics/metrics.go (hunk 1 above)
Comment (iulianpascalau) [reply node/metrics/metrics.go:78]: changed

**Review by bogdan-rosianu — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request closed without merging**